sdk/src/main/java/com/horizen/cryptolibprovider/CswCircuit.java

```diff
@@ -0,0 +1,52 @@
+package com.horizen.cryptolibprovider;
+
+import com.horizen.block.WithdrawalEpochCertificate;
+import com.horizen.box.Box;
+import com.horizen.librustsidechains.FieldElement;
+import com.horizen.proposition.Proposition;
+import com.horizen.secret.PrivateKey25519;
+import com.horizen.utils.ForwardTransferCswData;
+import com.horizen.utils.UtxoCswData;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface CswCircuit {
+    int utxoMerkleTreeHeight();
+
+    FieldElement getUtxoMerkleTreeLeaf(Box<Proposition> box);
+
+    byte[] getCertDataHash(WithdrawalEpochCertificate cert);
+
+    int rangeSize(int withdrawalEpochLength);
+
+    boolean generateCoboundaryMarlinSnarkKeys(int withdrawalEpochLen, String provingKeyPath, String verificationKeyPath);
+
+    byte[] privateKey25519ToScalar(PrivateKey25519 pk);
+
+    byte[] utxoCreateProof(UtxoCswData utxo,
+                           WithdrawalEpochCertificate lastActiveCert,
+                           byte[] mcbScTxsCumComEnd,
+                           byte[] senderPubKeyHash,
```

Comment:
Yes and no, as for me. That is the person who can spend the CSW coin in the MC as a CswInput. Because CSW is an input, it is more reasonable to call it "sender". Though  in design doc and sc-cryptolib it is called "receiver".  I can change to receiver, NP.